sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc

```diff
@@ -0,0 +1,284 @@
+The VNNI blocking factor is 2 in the case of 16bits, 4 in the case of 8 bits elements. While the current implementation assumes that the matrix has been already packed by the user for performance reasons, the layout information is needed to inform the implementation about this transform.
+
+
+      // Example of bf16 data type: 
+      // ---------------------------------
+      // a1, b1, c1, d1
+      // a2, b2, c2, d2
+      // a3, b3, c3, d3
+      // a4, b4, c4, d4
+      // ---------------------------------
+      // reformat to
+      // ---------------------------------
+      // a1, a2, b1, b2, c1, c2, d1, d2
+      // a3, a4, b3, b4, c3, c4, d3, d4
+
+
+## Example using int8_t type
+```c++
+using namespace sycl::ext::intel::experimental::matrix;
+
+queue q;
+range<2> G = {M, N};
+// For this first implementation, SG_SIZE has to be equal to one
+range<2> L = {1, SG_SIZE};
+int8_t *memA = malloc_shared<int8_t>(M*K, q);
+int8_t *memB = malloc_shared<int8_t>(K*N, q);
+Int32_t *memC = malloc_shared<int32_t>(M*N, q);
+//Assuming memB has already been VNNIed
+q.parallel_for(nd_range<2>(G, L), [=](nd_item<2> item)                            
+  [[sycl::reqd_sub_group_size(SG_SIZE)]] {
+   const auto global_idx = item.get_global_id(0);
+   const auto global_idy = item.get_global_id(1);
+   const auto sg_startx = global_idx - item.get_local_id(0);
+   const auto sg_starty = global_idy - item.get_local_id(1);
+   sub_group sg = item.get_sub_group();
+   joint_matrix<sub_group, int8_t, tM, tK> tA(sg);
+   // For B, since current implementation does not support non packed layout,
+   // users need to specify the updated VNNI sizes along with the packed_b layout
+   joint_matrix<sub_group, int8_t, tK/4, tN*4, packed_b> tB(sg);
+   joint_matrix<sub_group, int32_t, tM, tN> tC(sg);
+   joint_matrix_load(sg, tC, memC + sg_startx * tM * N + sg_starty, N, matrix_layout::row_major);
+   for (int k = 0; k < K; k += tk) {
+     joint_matrix_load(sg, tA, memA + sg_startx * tM * K + k, K, matrix_layout::row_major);//collective
+     joint_matrix_load(sg, tB, memB + k * N + sg_starty, N, matrix_layout::packed_b);//VNNI
+     tC = joint_matrix_mad(sg, tA, tB, tC);
+   }
+   joint_matrix_store(sg, tC, memC + sg_startx * tM * N + sg_starty, N, matrix_layout::row_major);
+}).wait();
+  
+```
+## Implementation Status
+For oneAPI release 3, an AOT implementation is available on the CPU device to targets AMX hardware. we are using AMX tile intrinsics to implement the matrix load and store operations. Since we are currently emitting AMX intrinsics directly, this only enables AOT compilation. Please refer to the following section that talks about the future unified SPIR-V path that will enable JIT compilation.
+// We used the https://software.intel.com/sites/landingpage/IntrinsicsGuide/#techs=AMX[`_tile_`-prefixed intrinsics] defined in `immintrin.h`.
```

Comment:
This is a comment line in markdown, so it doesn't show up in the rendered text.  The previous sentence says "Please refer to the following section", and I assume that means the section at this URL, right?  Therefore, I presume you do not want to comment out this line.

Reply:
> This is a comment line in markdown, so it doesn't show up in the rendered text.

Just to make sure... You are not using a MarkDown format in a file with an `.asciidoc` extension?

Reply:
Sorry, my mistake.  s/markdown/asciidoc/